test/03-device-state-v2.ts

```diff
@@ -248,8 +293,16 @@ describe('Device State v2', () => {
 
 			it(`Should see state become "offline" following a delay of ${(devicePollInterval +
 				TIMEOUT_MSEC) /
-				1000} seconds (plus 1 second to be sure)`, async () => {
-				await Bluebird.delay(devicePollInterval + TIMEOUT_MSEC + 1000);
+				1000} seconds`, async () => {
+				stateChangeEventSpy.resetHistory();
+
+				await Bluebird.delay(devicePollInterval + TIMEOUT_MSEC);
+
+				// it will be called for TIMEOUT and OFFLINE...
+				while (!stateChangeEventSpy.calledTwice) {
```

Comment:
It'd be good to have a max number of loops here rather than waiting forever on failure (and in the other while cases)